lib/payment.spec.js

```diff
@@ -58,58 +57,17 @@ chai.use(_chai => {
 });
 
 describe('Payment', () => {
-    const amount = '1000';
-    const currency = {
-        ct: '0x0000000000000000000000000000000000000001',
-        id: '0'
-    };
-    const recipient = '0x0000000000000000000000000000000000000003';
-    const privateKey = '3a1076bf45ab87712ad64ccb3b10217737f7faacbf2872e88fdd9a537d8fe266';
-    const sender = prefix0x(privateToAddress(Buffer.from(privateKey, 'hex')).toString('hex'));
-    let unsignedPayload, signedPayload;
-
-    const hashPaymentAsWallet = payload => {
-        const amountCurrencyHash = hash(
-            payload.amount,
-            payload.currency.ct,
-            payload.currency.id
-        );
-        const senderHash = hash(
-            payload.sender.wallet,
-            payload.sender.data,
-        );
-        const recipientHash = hash(
-            payload.recipient.wallet
-        );
-
-        return hash(amountCurrencyHash, senderHash, recipientHash);
-    };
-
-    beforeEach(() => {
-        unsignedPayload = {
-            amount,
-            currency,
-            sender: {
-                wallet: sender,
-                data: 'data'
-            },
-            recipient: {
-                wallet: recipient
-            }
-        };
-        signedPayload = {
-            ...unsignedPayload,
-            seals: {
-                wallet: {
-                    hash: hashPaymentAsWallet(unsignedPayload),
-                    signature: {
-                        r: '0x72e01a17545dfdd2cd5c7f34d7459f71579aad968cec0026fe34d1ae59168346',
-                        s: '0x767ac6c2d6e02bd092689c202e2f4935ca5779ac6e5c183f0e0d7095e01b1702',
-                        v: 27
-                    }
-                }
-            }
-        };
+    const senderRef = 'c3cc73cc-756a-11e9-957c-705ab6aee958';
+    let fixture;
+
+    beforeEach(function() {
+        fixture = new ApiPayloadFactory({
```

Comment:
Nice! Much more compact compared to the original implementation.

Reply:
First iteration of this kind of extracted test utils. Have many ideas for improvements, but the *main* benefit is already achieved: a single place to define what the scheme is during a test.